samples/sap-java-buildpack-tokenflow-usage/README.md

```diff
@@ -0,0 +1,147 @@
+# Description
+This sample uses the SAP application router as a web server and forwards requests to a Java back-end application running
+on the SAP Java buildpack. On incoming requests it reads credentials from the `VCAP_SERVICES` environment variable and
+requests a new access token via client credentials token flow provided by the [Token Client](../../token-client/)
+library.
+
+In a typcal UI5 application, the application router server HTML files and REST data would be provided by a 
+back-end application. To focus on the security part, UI5 has been omitted.
+
+# Coding
+The [web.xml](src/main/webapp/WEB-INF/web.xml) of the application must use auth-method with value XSUAA. 
+This enables authentication of requests using incoming OAuth authentication tokens.
+
+```xml
+<web-app>
+<display-name>sample</display-name>
+  <login-config> 
+    <auth-method>XSUAA</auth-method>
+  </login-config> 
+</web-app> 
+```
+
+In the Java coding, use the `@ServletSecurity` annotations:
+```java
+import com.sap.cloud.security.xsuaa.client.ClientCredentials;
+import com.sap.cloud.security.xsuaa.client.DefaultOAuth2TokenService;
+import com.sap.cloud.security.xsuaa.client.OAuth2TokenResponse;
+import com.sap.cloud.security.xsuaa.client.XsuaaDefaultEndpoints;
+import com.sap.cloud.security.xsuaa.tokenflows.XsuaaTokenFlows;
+import org.json.JSONObject;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.HttpConstraint;
+import javax.servlet.annotation.ServletSecurity;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/hello-tokenflow")
+@ServletSecurity(@HttpConstraint(rolesAllowed = { "Display" }))
+public class HelloTokenFlowServlet extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse
+	 * response)
+	 */
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+		response.setContentType("text/plain");
+
+		JSONObject jsonObject = createJsonObjectFromVCAPServices();
+		String clientSecret = extractString(jsonObject, "/xsuaa/0/credentials/clientsecret");
+		String clientid = extractString(jsonObject, "/xsuaa/0/credentials/clientid");
+		String url = extractString(jsonObject, "/xsuaa/0/credentials/url");
+
+		XsuaaTokenFlows tokenFlows = new XsuaaTokenFlows(
+				new DefaultOAuth2TokenService(),
+				new XsuaaDefaultEndpoints(url), new ClientCredentials(clientid, clientSecret));
+		OAuth2TokenResponse tokenResponse = tokenFlows.clientCredentialsTokenFlow().execute();
+
+		writeLine(response, "Access-Token: " + tokenResponse.getAccessToken());
+		writeLine(response, "Expired-At: " + tokenResponse.getExpiredAtDate());
```

Comment:
additionally you can return / print tokenResponse.getDecodedAccessToken.getPayload() ....